src/parse.rs

```diff
@@ -187,8 +206,17 @@ impl<'a> Parser<'a> {
                 self.index += 4;
                 Ok(())
             }
-            Some(_) => json_err!(InvalidNull, self.index),
-            None => json_err!(UnexpectedEnd, self.data.len()),
+            Some(_) | None => {
+                // determine the error position to match serde
+                self.index += 1;
+                for c in NULL_REST.iter() {
+                    if !self.data.get(self.index + 1).is_some_and(|v| v == c) {
+                        break;
+                    }
+                    self.index += 1;
+                }
+                json_err!(ExpectedIdent, self.index)
+            }
```

Comment:
This change probably wants to be applied to `true` and `false` also

Reply:
done.